discovery/targetgroup/targetgroup.go

```diff
@@ -91,3 +91,26 @@ func (tg *Group) UnmarshalJSON(b []byte) error {
 	tg.Labels = g.Labels
 	return nil
 }
+
+// Equal returns true if all fields have exactly the same values and lenghts.
+// To keep a good performance the `Targets []model.LabelSet` is not sorted
+// so if these arrive in a different order the comparing won't work.
```

Comment:
I think "if .Targets of the compared objects are of different order they are not equal", would be clearer, but I'm also unsure whether this semantic might be a problem, as I don't think we ask provider implementations to have this consistent, just some (like Kubernetes) happen to report targets in a consistent order.

Reply:
another way to  deal with this is to filter these in the k8s provider code, but not sure if another provider might misbehave in the same way.

Reply:
Filtering them would be incorrect, the events rightfully end up here, it just happens to be a side effect, it could just as well be a valid update.

Reply:
`SelfLink:/api/v1/namespaces/kube-system/endpoints/kube-controller-manager`

can this actually include some targets to be scraped? if yes than end of discussion can't filter these.

About the slice order, implementing to order the maps would mean many iterations in a loop so it would be probably worse than just allowing the duplicate updates.